terraspace/app/modules/lambda/auth.tf

```diff
@@ -18,11 +18,12 @@ EOF
 }
 
 resource "aws_lambda_permission" "with_sns" {
+  for_each      = { for topic in var.sns_topic_trigger_arns : topic => topic }
```

Comment:
Could you check to see what, if any, difference there is with this versus `for_each = toset(var.sns_topic_trigger_arns)`

Reply:
toset is better here. This complex structure only makes sense when handling object properties.